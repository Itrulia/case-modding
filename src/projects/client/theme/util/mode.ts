import styledTheme from "styled-theming";
import { themeFunction } from "./helpers";

export interface ModeValues<T> {
    dark: T;
}

export type ThemeMode = keyof ModeValues<any>;

export interface HasMode {
    mode: ThemeMode;
}

export const createMode = (
    key: keyof HasMode,
    obj: ModeValues<string | themeFunction>
) => styledTheme(key, obj);
