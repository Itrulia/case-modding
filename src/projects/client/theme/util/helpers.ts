import * as styledComponents from "styled-components";
import { ThemedStyledComponentsModule } from "styled-components";
import { HasMode } from "./mode";
import { HasMood } from "./mood";

const {
    default: styled,
    css,
    keyframes,
    ThemeProvider,
    createGlobalStyle,
    ThemeContext
} = styledComponents as ThemedStyledComponentsModule<ThemeInterface>;

export interface TypographyScale {
    lineHeight: number;
    fontSize: number;
}

export interface ThemeInterface extends HasMode, HasMood {
    primaryHue: number;
    secondaryHue: number;
    // states
    dangerHue: number;
    warningHue: number;
    successHue: number;

    gridSpan: number;
}

export {
    css,
    styled,
    keyframes,
    ThemeProvider,
    createGlobalStyle,
    ThemeContext
};

export interface Props {
    theme: ThemeInterface;
}

export type themeFunction = (props: Props) => string;
