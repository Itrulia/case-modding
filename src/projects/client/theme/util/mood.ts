import styledTheme from "styled-theming";
import { themeFunction } from "./helpers";

export interface MoodValues<T> {
    playful: T;
}

export type ThemeMood = keyof MoodValues<any>;

export interface HasMood {
    mood: ThemeMood;
}

export const createMood = (
    key: keyof HasMood,
    obj: MoodValues<string | themeFunction>
) => styledTheme(key, obj);
