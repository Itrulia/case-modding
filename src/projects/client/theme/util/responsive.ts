import { css, ThemeInterface } from "./helpers";
import { ThemedCssFunction } from "styled-components";

export const createQuery = (
    minSize: number,
    maxSize?: number
): ThemedCssFunction<ThemeInterface> => (strings, ...interpolations) => {
    if (maxSize) {
        return css`
            @media (min-width: ${minSize / 16}em) and (max-width: ${maxSize /
                    16}em) {
                ${css(strings, ...interpolations)};
            }
        `;
    }

    return css`
        @media (min-width: ${minSize / 16}em) {
            ${css(strings, ...interpolations)};
        }
    `;
};
