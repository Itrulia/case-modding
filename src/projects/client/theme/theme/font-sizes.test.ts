import { ThemeInterface } from "../util/helpers";
import {
    h1FontSize,
    h1LineHeight,
    h2FontSize,
    h2LineHeight,
    h3FontSize,
    h3LineHeight,
    h4FontSize,
    h4LineHeight,
    h5FontSize,
    h5LineHeight,
    h6FontSize,
    h6LineHeight,
    bodyFontSize,
    bodyLineHeight
} from "./font-sizes";

const theme = {} as ThemeInterface;

describe("h1FontSize", () => {
    it("matches mobile snapshot", () => {
        const result = h1FontSize({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = h1FontSize({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = h1FontSize({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});

describe("h1LineHeight", () => {
    it("matches mobile snapshot", () => {
        const result = h1LineHeight({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = h1LineHeight({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = h1LineHeight({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});

describe("h2FontSize", () => {
    it("matches mobile snapshot", () => {
        const result = h2FontSize({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = h2FontSize({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = h2FontSize({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});

describe("h2LineHeight", () => {
    it("matches mobile snapshot", () => {
        const result = h2LineHeight({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = h2LineHeight({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = h2LineHeight({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});

describe("h3FontSize", () => {
    it("matches mobile snapshot", () => {
        const result = h3FontSize({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = h3FontSize({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = h3FontSize({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});

describe("h3LineHeight", () => {
    it("matches mobile snapshot", () => {
        const result = h3LineHeight({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = h3LineHeight({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = h3LineHeight({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});

describe("h4FontSize", () => {
    it("matches mobile snapshot", () => {
        const result = h4FontSize({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = h4FontSize({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = h4FontSize({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});

describe("h4LineHeight", () => {
    it("matches mobile snapshot", () => {
        const result = h4LineHeight({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = h4LineHeight({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = h4LineHeight({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});

describe("h5FontSize", () => {
    it("matches mobile snapshot", () => {
        const result = h5FontSize({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = h5FontSize({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = h5FontSize({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});

describe("h5LineHeight", () => {
    it("matches mobile snapshot", () => {
        const result = h5LineHeight({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = h5LineHeight({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = h5LineHeight({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});

describe("h6FontSize", () => {
    it("matches mobile snapshot", () => {
        const result = h6FontSize({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = h6FontSize({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = h6FontSize({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});

describe("h6LineHeight", () => {
    it("matches mobile snapshot", () => {
        const result = h6LineHeight({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = h6LineHeight({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = h6LineHeight({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});

describe("bodyFontSize", () => {
    it("matches mobile snapshot", () => {
        const result = bodyFontSize({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = bodyFontSize({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = bodyFontSize({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});

describe("bodyLineHeight", () => {
    it("matches mobile snapshot", () => {
        const result = bodyLineHeight({ theme: { ...theme, size: "mobile" } });
        expect(result).toMatchSnapshot();
    });

    it("matches tablet snapshot", () => {
        const result = bodyLineHeight({ theme: { ...theme, size: "tablet" } });
        expect(result).toMatchSnapshot();
    });

    it("matches desktop snapshot", () => {
        const result = bodyLineHeight({ theme: { ...theme, size: "desktop" } });
        expect(result).toMatchSnapshot();
    });
});
