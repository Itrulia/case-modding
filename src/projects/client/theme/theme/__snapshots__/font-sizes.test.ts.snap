// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bodyFontSize matches desktop snapshot 1`] = `undefined`;

exports[`bodyFontSize matches mobile snapshot 1`] = `undefined`;

exports[`bodyFontSize matches tablet snapshot 1`] = `undefined`;

exports[`bodyLineHeight matches desktop snapshot 1`] = `undefined`;

exports[`bodyLineHeight matches mobile snapshot 1`] = `undefined`;

exports[`bodyLineHeight matches tablet snapshot 1`] = `undefined`;

exports[`h1FontSize matches desktop snapshot 1`] = `undefined`;

exports[`h1FontSize matches mobile snapshot 1`] = `undefined`;

exports[`h1FontSize matches tablet snapshot 1`] = `undefined`;

exports[`h1LineHeight matches desktop snapshot 1`] = `undefined`;

exports[`h1LineHeight matches mobile snapshot 1`] = `undefined`;

exports[`h1LineHeight matches tablet snapshot 1`] = `undefined`;

exports[`h2FontSize matches desktop snapshot 1`] = `undefined`;

exports[`h2FontSize matches mobile snapshot 1`] = `undefined`;

exports[`h2FontSize matches tablet snapshot 1`] = `undefined`;

exports[`h2LineHeight matches desktop snapshot 1`] = `undefined`;

exports[`h2LineHeight matches mobile snapshot 1`] = `undefined`;

exports[`h2LineHeight matches tablet snapshot 1`] = `undefined`;

exports[`h3FontSize matches desktop snapshot 1`] = `undefined`;

exports[`h3FontSize matches mobile snapshot 1`] = `undefined`;

exports[`h3FontSize matches tablet snapshot 1`] = `undefined`;

exports[`h3LineHeight matches desktop snapshot 1`] = `undefined`;

exports[`h3LineHeight matches mobile snapshot 1`] = `undefined`;

exports[`h3LineHeight matches tablet snapshot 1`] = `undefined`;

exports[`h4FontSize matches desktop snapshot 1`] = `undefined`;

exports[`h4FontSize matches mobile snapshot 1`] = `undefined`;

exports[`h4FontSize matches tablet snapshot 1`] = `undefined`;

exports[`h4LineHeight matches desktop snapshot 1`] = `undefined`;

exports[`h4LineHeight matches mobile snapshot 1`] = `undefined`;

exports[`h4LineHeight matches tablet snapshot 1`] = `undefined`;

exports[`h5FontSize matches desktop snapshot 1`] = `undefined`;

exports[`h5FontSize matches mobile snapshot 1`] = `undefined`;

exports[`h5FontSize matches tablet snapshot 1`] = `undefined`;

exports[`h5LineHeight matches desktop snapshot 1`] = `undefined`;

exports[`h5LineHeight matches mobile snapshot 1`] = `undefined`;

exports[`h5LineHeight matches tablet snapshot 1`] = `undefined`;

exports[`h6FontSize matches desktop snapshot 1`] = `undefined`;

exports[`h6FontSize matches mobile snapshot 1`] = `undefined`;

exports[`h6FontSize matches tablet snapshot 1`] = `undefined`;

exports[`h6LineHeight matches desktop snapshot 1`] = `undefined`;

exports[`h6LineHeight matches mobile snapshot 1`] = `undefined`;

exports[`h6LineHeight matches tablet snapshot 1`] = `undefined`;
